packages/office-ui-fabric-react/src/utilities/decorators/withResponsiveMode.tsx

```diff
@@ -59,48 +59,42 @@ export function withResponsiveMode<TProps extends { responsiveMode?: ResponsiveM
 ): any {
   const resultClass = class WithResponsiveMode extends BaseDecorator<TProps, IWithResponsiveModeState> {
     private _events: EventGroup;
+    private _async: Async;
 
     constructor(props: TProps) {
       super(props);
       this._events = new EventGroup(this);
+      this._async = new Async(this);
       this._updateComposedComponentRef = this._updateComposedComponentRef.bind(this);
 
-      this.state = {
-        responsiveMode: getInitialResponsiveMode(),
-      };
+      // prevent excessive rerenders by using debounce
+      this._onResize = this._async.debounce(this._onResize, 500);
     }
 
     public componentDidMount(): void {
       this._events.on(window, 'resize', this._onResize);
-      this._onResize();
     }
 
     public componentWillUnmount(): void {
       this._events.dispose();
     }
 
     public render(): JSX.Element | null {
-      const { responsiveMode } = this.state;
-
-      return responsiveMode === ResponsiveMode.unknown ? null : (
-        <ComposedComponent
-          ref={this._updateComposedComponentRef}
-          responsiveMode={responsiveMode}
-          {...(this.props as any)}
-        />
+      return (
+        <WindowContext.Consumer>
+          {({ window }) => (
+            <ComposedComponent
+              ref={this._updateComposedComponentRef}
+              responsiveMode={getResponsiveMode(window)}
+              {...(this.props as any)}
+            />
+          )}
+        </WindowContext.Consumer>
       );
     }
 
     private _onResize = () => {
-      const element = findDOMNode(this) as Element;
-      const currentWindow = (element && getWindow(element)) || window;
-      const responsiveMode = getResponsiveMode(currentWindow);
-
-      if (responsiveMode !== this.state.responsiveMode) {
-        this.setState({
-          responsiveMode,
-        });
-      }
+      this.forceUpdate();
```

Comment:
SharePoint Online's top-level wrapper component is `withResponsiveMode` so we need to be cautious of introducing unnecessary forced renders. We've had performance problems with the component previously (see https://github.com/microsoft/fluentui/pull/12258).

How does this change look in the profiler before & after?

cc: @pingj

Reply:
could we keep the existing logic that only re-render when the mode changes. As @KevinTCoughlin  said, it is on the top level wrapper, and many sub components don't have shouldComponentUpdate, so they will just re-render.